lib/voyager_helpers/availability_status.rb

```diff
@@ -0,0 +1,51 @@
+module VoyagerHelpers
+  class AvailabilityStatus
+    attr_reader :availability_info
+    def initialize(availability_info)
+      @availability_info = availability_info
+    end
+
+    def status_label
+      hash_label || regex_label || status
+    end
+
+    def status
+      availability_info[:status]
+    end
+
+    private
+
+    def hash_label
+      Array(status_label_hash.find { |k, _| k.include?(status) }).last
+    end
+
+    def regex_label
+      Array(regex_hash.find { |k, _| status.match(k) }).last
+    end
+
+    def regex_hash
+      {
+        /On-site - / => 'See front desk',
+        /On-site/ => 'On-site access',
```

Comment:
The "on-site" statuses come from location-based logic that lives in [marc_liberation](https://github.com/pulibrary/marc_liberation/blob/a955313266e026c6c16764e75445a889126dc330/app/controllers/availability_controller.rb#L114-L122), so L28 and L29 won't have an effect.